runtime/src/iree/hal/drivers/metal/staging_buffer.m

```diff
@@ -35,14 +35,17 @@ iree_status_t iree_hal_metal_staging_buffer_initialize(
   out_staging_buffer->capacity = (uint32_t)buffer_capacity;
   out_staging_buffer->metal_buffer = metal_buffer;
   out_staging_buffer->host_buffer = metal_buffer.contents;
+  iree_slim_mutex_initialize(&out_staging_buffer->offset_mutex);
   out_staging_buffer->offset = 0;
-  out_staging_buffer->pending_command_buffers = 0;
+  iree_atomic_ref_count_init(&out_staging_buffer->pending_command_buffers);  // -> 1
+  iree_atomic_ref_count_dec(&out_staging_buffer->pending_command_buffers);   // -> 0
```

Comment:
`iree_atomic_ref_count_t` doesn't seem appropriate for this usage. This isn't tracking how many objects are referencing another object, it's tracking how many items are in a list...? How about just `iree_atomic_int32_t`? I guess I'm also a little confused now about `iree_hal_metal_staging_buffer_increase_refcount`...

Reply:
Yes it's tracking how many pending command buffers we have for the staging buffer. I see the confusion here. Let me actually rename to make it clear.

Reply:
> nicely for managing refcounts

If it's actually a refcount, which I'm not sure it is...

Could see if @benvanik has an opinion on what type to use and which functions to use with that type.

Reply:
Fair enough. I've changed to use the non-refcount version. :)